tests/src/test/spark350/scala/com/nvidia/spark/rapids/iceberg/data/GpuDeleteFilterSuite.scala

```diff
@@ -0,0 +1,496 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "350"}
+{"spark": "351"}
+{"spark": "352"}
+{"spark": "353"}
+{"spark": "354"}
+{"spark": "355"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.iceberg.data
+
+import scala.collection.JavaConverters._
+import scala.language.reflectiveCalls
+
+import ai.rapids.cudf.{HostColumnVector, HostColumnVectorCore}
+import com.nvidia.spark.rapids.{GpuColumnVector, LazySpillableColumnarBatch, NoopMetric, RapidsConf}
+import com.nvidia.spark.rapids.Arm.withResource
+import com.nvidia.spark.rapids.GpuMetric.{JOIN_TIME, OP_TIME}
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
+import com.nvidia.spark.rapids.iceberg.{fieldIndex, PooledTableGen}
+import com.nvidia.spark.rapids.iceberg.data.TestGpuDeleteLoader._
+import com.nvidia.spark.rapids.iceberg.parquet.{GpuIcebergParquetReaderConf, SingleFile}
+import com.nvidia.spark.rapids.spill.SpillFramework
+import org.apache.hadoop.conf.Configuration
+import org.apache.iceberg.{DeleteFile, FileContent, FileFormat, FileMetadata, MetadataColumns, PartitionSpec, Schema}
+import org.apache.iceberg.MetadataColumns.isMetadataColumn
+import org.apache.iceberg.common.DynMethods
+import org.apache.iceberg.spark.GpuTypeToSparkType.toSparkType
+import org.apache.iceberg.types.Type.TypeID
+import org.apache.iceberg.types.Types
+import org.apache.iceberg.types.Types.NestedField
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.prop.TableDrivenPropertyChecks._
+import org.scalatest.prop.Tables.Table
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.types.{DataType, LongType, StringType, StructType}
+import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
+
+class GpuDeleteFilterSuite extends AnyFunSuite with BeforeAndAfterAll {
+  override def beforeAll(): Unit = {
+    SpillFramework.initialize(new RapidsConf(new SparkConf))
+  }
+
+  override def afterAll(): Unit = {
+    SpillFramework.shutdown()
+  }
+
+  private def fixture(deleteFiles: Seq[DeleteFile], _tableSchema: Schema = TABLE_SCHEMA) = new {
+    val tableSchema = _tableSchema
+    val deletedRows: Map[String, Seq[Long]] = Map(
+      PooledTableGen.PooledFilePaths(0) -> Seq(1L, 3L, 5L),
+      PooledTableGen.PooledFilePaths(1) -> Seq(2L, 4L, 7L),
+    )
+    val tableGen = new PooledTableGen(tableGenSchema(tableSchema, deleteFiles))
+    val deleteLoader = new TestGpuDeleteLoader(tableGen, 57, deletedRows)
+    val deleteFilter = gpuDeleteFilterOf(tableSchema, deleteFiles, Some(deleteLoader))
+    val generatedInputData = {
+      val batch1 = () => tableGen.toColumnarBatch(NUM_ROWS)
```

Comment:
Nit: No need to use lambda.
 val batch1 = tableGen.toColumnarBatch(NUM_ROWS)
 ......
then: 
(batch1, batch2, batch3)